juju/juju
# add juju abort command

## Description of change

Add the juju abort command for branches and related tests.  Includes an update
the modelgeneration package in api.

## QA steps

1. export JUJU_DEV_FEATURE_FLAGS=generations
2. juju bootstrap
2. juju deploy wordpress
3. juju add-branch apple
2. juju track apple wordpress 
2. juju abort apple             <- fails due to assigned units.
4. juju commit apple
3. juju add-branch banana
3. juju config wordpress debug=yes --branch banana
3. juju abort banana     <- succeeds

Verify in the juju db,"generation-id" : 0, and completed and completed-by are set.

Please try other permutations as well.
## Documentation changes

yes
